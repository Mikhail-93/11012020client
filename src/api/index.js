import 'isomorphic-fetch'
import fetchIntercept from "fetch-intercept/lib/node";

import errorInterceptor from "./errorInterceptor";
import dataInterceptor from "./dataInterceptor";
import cookieInterceptor from "./cookieInterceptor";
import headersInterceptor from "./headersInterceptor";
import api from "./api";

fetchIntercept.register(errorInterceptor)
fetchIntercept.register(dataInterceptor)
fetchIntercept.register(cookieInterceptor)
fetchIntercept.register(headersInterceptor)



export default api
