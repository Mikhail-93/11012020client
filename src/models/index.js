/**
 * Created by alexey2baranov on 8/20/16.
 */

import AbstractSync from "./AbstractSync" //для того чтобы правильно работали рекурсивные ссылки модулей
// import Registration from "./model/Registration"
import Kopnik from "./Kopnik"
// import Kopa from "./model/Kopa"
// import File from "./model/File"

export {AbstractSync, Kopnik}