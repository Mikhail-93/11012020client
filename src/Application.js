import {AbstractSync, Kopnik} from "./models";

export default class Application {
    static getInstance() {
        if (!this.instance) {
            this.instance = new this()
        }
        return this.instance
    }

    user = null

    constructor() {
        /**
         * Кэш моделей
         * @type {Array}
         */
        // пока не понятно зачем приложению кэш.
        // он вероятно будет сильно торможить приложение из-за реактивности App.$data в которой он находится
        // хотя он же не в App.$data а в App.$options !!
        // уже в дата потому тчо в options не реактивен
        //this.models = AbstractSync.cache

        /**
         * Идентификатор раздела
         * Равен названиею соответствующего компонента
         *
         * @type {string} Map | Profile | Thanks
         */
        this.SECTION = "Map"

        this.user = undefined

        this.initUser()
    }

    getSharedState() {
        return {
            SECTION: this.SECTION
        }
    }

    setState(state) {
        if (state.SECTION) {
            this.SECTION = state.SECTION
        }
    }

    /**
     * Инициализирует ползователя при запуске приложения
     *
     * @returns {Promise<void>}
     */
    async initUser() {
        let vkUser = localStorage.getItem("vkUser")
        if (vkUser) {
            vkUser = JSON.parse(vkUser)
            let user //= await Kopnik.getVkUserStatus(vkUser.uid)
            if (vkUserStatus) {
                this.user= Kopnik.merge(vkUserStatus.user)
            } else {
                this.setVkUser(vkUser)
                this.SECTION="Profile"
            }
        }
    }

    /**
     * Учстанавливает пользователем объект из VK.init::success()
     * @param vkUser
     * @returns
     */
    setVkUser(vkUser){
        this.user= new Kopnik
        this.user.merge({
            hash: vkUser.hash,
            uid: vkUser.uid,
            firstname: vkUser.first_name,
            surname: vkUser.last_name,
            photo: vkUser.photo,
            smallPhoto: vkUser.photo_rec,
        })
        localStorage.setItem("vkUser", JSON.stringify(vkUser))
    }
}
