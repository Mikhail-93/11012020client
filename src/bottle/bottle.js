import Bottle from "bottlejs"
import _ from "lodash"
// import 'json-form-data'
import * as loglevel from "loglevel";

import Application from "../application/Application"
import constants from './constants'
import api from "../api";
import CookieService from "./CookieService"
import MK from "../mk/mk";
import LocaleManager from "../locales/LocaleManager";
import ru from '../locales/ru'
import en from '../locales/en'

Bottle.config.strict = true
const bottle = new Bottle()

bottle.factory('logger', function loggerFactory() {
    //все плагины ломают стектрейс консоли. то есть невозможно увидеть из какого файла и какой строки был вызван лог!
    // LoglevelPluginPrefix.reg(loglevel)
    // LoglevelPluginPrefix.apply(loglevel, {template: "%t [%l] %n: "})
    loglevel.setLevel(loglevel.levels.DEBUG)
    if (process.env.NODE_ENV == "test") {
        loglevel.setLevel(loglevel.levels.DEBUG)
        // LoglevelPluginToString.apply(loglevel, {})
    }
// Be sure to call setLevel method in order to apply plugin
// loglevel.getLogger("StateManager").setLevel("info")
    return loglevel
})

bottle.constant('messages', {
    ru,
    en,
})

bottle.factory('env', function env() {
    return process.env.NODE_ENV || 'development'
})
bottle.service('localeManager', function localeManager() {
    const localeManager = new LocaleManager()
    localeManager.currentLocale = 'ru'//container.env === 'development' ? 'en' : 'ru'
    return localeManager
})
bottle.service('cookieService', CookieService, 'constants')
bottle.factory('api', function apiFactory(container) {
    if (!container.constants.di.fetch) {
        throw new Error('not supported')
    }
    return api
})
bottle.factory('VK', function vkFactory(container) {
    return process.env.NODE_ENV === 'test' ? MK : global.VK
})
bottle.factory('constants', function constantsFactory() {
    if (!process.env.NODE_ENV) {
        throw new Error("NODE_ENV is not defined")
    }

    let local = {}//require("./local.js")
    let result = _.merge({}, constants, local)[process.env.NODE_ENV]

    return result
})
if (!Application) {
    throw new Error("Application is undefined in bottle")
}
bottle.service('application', Application, 'logger')

/**
 * @callback fetch
 * @param {string} url
 * @param {Object?} options
 */
/**
 * @type {Object}
 *
 * @property {loglevel} logger
 * @property {Application} application
 * @property {Location} Location
 * @property {fetch} api
 * @property {MK.class} VK
 * @property {CookieService} cookieService
 * @property {Object} constants
 * @property {Object} constants.api
 * @property {String} constants.api.path
 * @property {Object} constants.di
 * @property {Boolean} constants.di.fetch
 * @property {Number} constants.di.cookie
 * @property {Number} constants.messenger.clientId
 * @property {String} constants.messenger.loginUrl
 * @property {String} constants.messenger.redirectUrl
 * @property {String} constants.sw
 * @property {String} constants.sw.delay
 * @property {LocaleManager} localeManager
 * @property {string} env
 * @property {{ru, en}} messages
 */
const container = bottle.container
export {bottle, container}
