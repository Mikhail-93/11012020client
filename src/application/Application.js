import get from "keypather/get"
import has from "keypather/has"

import AsyncLock from 'async-lock'
import {AbstractSync, Kopnik, Kopa} from "../models";
import {KopnikApiError, KopnikError} from "../KopnikError";
import once from "../decorators/once";
import SquadAnalyzer from "../SquadAnalyzer";
import fetchIntercept from 'fetch-intercept'
import {container} from "../bottle/bottle";
import {LatLng, LatLngBounds} from 'leaflet'
import Main from "./Main";
import messages from "../locales";

export default class Application {
    constructor(logger) {
        this.logger = logger.getLogger('Application')
        /**
         * Кэш моделей
         * @type {Array}
         */
        // пока не понятно зачем приложению кэш.
        // он вероятно будет сильно торможить приложение из-за реактивности App.$data в которой он находится
        // хотя он же не в App.$data а в App.$options !!
        // уже в дата потому тчо в options не реактивен
        //this.models = AbstractSync.cache

        /**
         * Идентификатор раздела
         * Равен названиею соответствующего компонента
         *
         * @type {string} Map | Profile | Thanks
         */
        this.section = Application.Section.Main
        this.sectionLocker = new AsyncLock

        /**
         * 0 потому что на undefined не срабатывает watcher,
         * а null занят под "пользователь не аутентифицирован"
         * @type {Kopnik}
         */
        this.user = undefined

        /**
         * Сообщения пользователю
         * @type {String[]}
         */
        this.infos = []
        /**
         * Обишки приложения
         * @type {Error[]}
         */
        this.errors = []
        this.sections = {
            main: new Main(this)
        }
    }

    async onerror(err) {
        if (err.code === 401) {
            await container.application.lockSection(async () => {
                await application.setSection(Application.Section.Main)
            })
            this.user= null
            console.info('prevent 401 error', err)
        } else {
            this.logger.error(err)
            if (err instanceof KopnikApiError){
                this.logger.info(err.url)
            }
            this.errors.push(err)
            // throw err
        }
    }

    getMessage(message) {
        for(const eachLocaleName of [container.localeManager.currentLocale.name, 'ru']) {
            if (has(container.messages[eachLocaleName], message)) {
                const result = get(container.messages[eachLocaleName], message)
                return result
            }
        }
        throw new KopnikError('No message provided: ' + message, 2)
    }

    /**
     * Эксклюзивный доступ к section
     * @param {function} callback
     * @returns {Promise<any>}
     */
    lockSection(callback) {
        return this.sectionLocker.acquire('section', callback)
    }

    /**
     * Устанавливить секцию
     * Если секция не может быть установлена, возвращает иную
     *
     * @param {String} section
     * @returns {Promise<string>} Установленная секция
     */
    async setSection(section) {
        let result
        if (this.section === section) {
            return section
        }
        switch (section) {
            case Application.Section.Profile:
            case Application.Section.Witness:
            case Application.Section.Ten:
                if (await this.resolveUser()) {
                    if (section === Application.Section.Profile || this.user.status === Kopnik.Status.CONFIRMED) {
                        result = section
                    } else {
                        result = await this.setSection(Application.Section.Main)
                    }
                } else {
                    result = await this.setSection(Application.Section.Main)
                }
                break
            case Application.Section.Main:
            case Application.Section.Thanks:
                result = section
                break
            default:
                throw new KopnikError('Wrong route', 666)
        }
        // this.logger.info('move', this.section, '->', section)
        return this.section = result
    }


    getSharedState() {
        return {
            SECTION: this.section
        }
    }

    setState(state) {
        if (state.section) {
            this.section = state.section
        }
    }

    /**
     * Определиться с пользователем. Или он некий копник, или он null, то есть незарегистрирован на сервере
     * @returns {Promise<Kopnik>}
     */
    async resolveUser() {
        if (this.user === undefined) {
            await this.authenticate()
        }
        return this.user
    }

    /**
     * Инициализирует ползователя при запуске приложения
     *
     * @returns {Promise<void>}
     */
    @once
    async authenticate() {
        try {
            let userAsPlain = (await Kopnik.api('get?ids='))[0]
            this.user = await Kopnik.get(userAsPlain.id)
            container.localeManager.currentLocale= this.user.locale
            this.logger.info('user authenticated', this.user)
        } catch (err) {
            if ((err instanceof KopnikApiError) && err.message.match(/no.+aut/i)) {
                 this.user = null
                this.logger.info('user not authenticated')
            } else {
                throw err
            }
        }
    }

    /**
     * Инициализирует ползователя при запуске приложения
     *
     * @returns {Promise<void>}
     */
    @once
    async logout() {
        await this.lockSection(() => {
            this.setSection(Application.Section.Main)
        })
        this.sections.main.selected= null
        await container.api('logout')
        this.user = null
    }
}

/**
 * @enum {string}
 */
Application.Section = {
    Main: 'Main',
    Profile: 'Profile',
    Witness: 'Witness',
    Thanks: 'Thanks',
    Ten: 'Ten',
    Help: 'Help',
}
