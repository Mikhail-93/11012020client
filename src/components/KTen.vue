<template>
    <v-container fluid class="fill-height k-ten flex-column align-center">
        <v-card elevation="12" class="mb-10" width="100%" max-width="350px">
            <!--            <v-card-title>
                            Моя десятка
                        </v-card-title>-->
            <div class="d-flex flex-wrap mt-2 justify-space-around">
                <template v-for="(eachDruzhe, eachDruzheIndex) of extendedTen">
                    <v-badge v-if="eachDruzhe" :key="eachDruzhe.id" :title="eachDruzhe.name"
                             class='k-badge-event-handler' color="red" :offset-x="(64/64)*7+14" :offset-y="(64/64)*7+14"
                    >
                        <div @click="druzhe_del(eachDruzhe)" slot="badge" style="cursor: pointer">x</div>
                        <AvatarVue :value="eachDruzhe"
                                   :size="64" class="mb-2">
                        </AvatarVue>
                    </v-badge>
                    <v-avatar v-else :key="-eachDruzheIndex"

                 :size="64" class="mb-2">
                        <v-icon :size="64">mdi-account</v-icon>
                    </v-avatar>
                </template>
            </div>
        </v-card>

        <transition-group v-if="application.user.tenRequests && application.user.tenRequests.length" name="list-complete" tag="div" class="mx-auto"
                          style="width: 100%; max-width:350px; position: relative;">
            <v-card v-for="eachRequest in value.tenRequests" :key="eachRequest.id"
                    elevation="12" class="mb-10 list-complete-item" style="width: 100%;">
                <kopnik-view :value="eachRequest" location></kopnik-view>
                <v-divider></v-divider>
                <v-card-actions>
                    <v-btn color="success" class="flex"
                           @click="request_confirm(eachRequest)">
                        Принять в десятку
                    </v-btn>
                    <v-btn color="error" class="flex"
                           @click="request_decline(eachRequest)">
                        Отклонить
                    </v-btn>
                </v-card-actions>
            </v-card>
        </transition-group>
        <div v-else class="overline font-weight-bold my-auto" style="font-size: 1.625rem !important; color: #BDBD">Заявок нет</div>
    </v-container>
</template>
<script>
    import Kopnik from "../models/Kopnik"
    import thanks from "../thanks";

    import KopnikView from './KopnikVue'
    import AvatarVue from "./AvatarVue";
    import logger from "./mixin/logger";
    import {container} from "../plugins/bottle";

    export default {
        name: "Ten",
        mixins: [logger],
        components: {
            KopnikView,
            AvatarVue,
        },
        data: () => {
            return {
                application: container.application,
                items: [1, 2, 3, 4, 5, 6, 7, 8, 9],
            }
        },
        props: {
            value: {
                type: Kopnik,
                required: true
            }
        },
        computed: {
            extendedTen() {
                const result = [],
                    ten = this.application.user.ten
                for (let length = 0; length < 10; length++) {
                    result.push((ten !== undefined && ten.length >= length) ? ten[length] : null)
                }
                return result
            }
        },
        methods: {
            shuffle: function () {
                this.items.splice(0, 1)
                // this.items = _.shuffle(this.items)
            },
            request_confirm(request) {
                this.shuffle()
                const requests = this.application.user.tenRequests
                requests.splice(requests.indexOf(request), 1)
                this.application.user.ten.push(request)
            },
            request_decline(request) {
                const requests = this.application.user.tenRequests
                requests.splice(requests.indexOf(request), 1)
            },
            druzhe_del(druzhe) {
                const ten = this.application.user.ten
                ten.splice(ten.indexOf(druzhe), 1)
            }
        }
        ,
        async created() {
            // this.application.user.ten = [Kopnik.getReference(1), Kopnik.getReference(3), Kopnik.getReference(4)]
            this.application.user.tenRequests = [
                Kopnik.getReference(1), Kopnik.getReference(3), Kopnik.getReference(4),
            ]
        }
        ,
    }
</script>

<style>
    .k-badge-event-handler .v-badge__wrapper{
        pointer-events: all;
    }

    .list-complete-leave-active {
        position: absolute;
    }

    .list-complete-leave-to {
        opacity: 0;
        transform: translateX(100px);
    }

    .list-complete-item {
        transition: all .25s;
        display: inline-block;
        margin-right: 10px;
    }
</style>
