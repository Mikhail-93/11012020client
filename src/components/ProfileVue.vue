<template>
    <v-container v-if="request"
                 fluid class="fill-height">
        <ValidationObserver ref="obs" v-slot="{ invalid, validated, passes, validate }"
                            tag="div" class="mx-auto" style="width: 100%; max-width:350px">
            <v-card elevation="12">
                <v-card-text>
                    <v-form>
                        <kopnik-vue :value="request"
                                    locale fio birthyear passport role location
                                    @locale_change="locale_change"
                                    @map_move="map_move"
                        ></kopnik-vue>
                        <v-list v-if="isMessagesFromGroupAllowed===false">
                            <v-list-item>
                                <v-list-item-content>
                                    <v-list-item-title style="white-space: inherit !important;">{{
                                        $t('profile.messagesFromGroup.allow') }}
                                    </v-list-item-title>
                                    <!-- VK Widget https://vk.com/dev/AllowMessagesFromCommunity https://vk.com/dev/widget_allow_messages_from_community -->
                                    <div id="vk_allow_messages_from_community" class="text-center my-3"></div>
                                </v-list-item-content>
                            </v-list-item>
                            <slot></slot>
                        </v-list>
                        <v-btn id='submit' color="primary" block
                               :disabled="invalid || !isMessagesFromGroupAllowed || !request.location.lat"
                               @click="submit_click"
                        >
                            {{$t('profile.submit')}}
                        </v-btn>
                    </v-form>
                </v-card-text>
            </v-card>
        </ValidationObserver>
    </v-container>
</template>
<script>
    import {
        ValidationObserver,
        localize
    } from "vee-validate"

    import {Kopnik} from "../models"
    import {container} from "../bottle/bottle";
    import logger from "./mixin/logger";
    import KopnikVue from "./KopnikVue";

    export default {
        mixins: [logger],
        name: "Profile2",
        components: {
            ValidationObserver,
            KopnikVue,
        },

        data() {
            return {
                application: container.application,
                request: null,
                // в текущий момент
                isMessagesFromGroupAllowed: undefined,
                // при инициализации страницы
                wasMessagesFromGroupAllowed: undefined,
            }
        },
        props: {},
        computed: {},
        watch: {},
        methods: {
            map_move(event) {
                this.request.location = event.target.getCenter()
                // console.log(event.target.getCenter())
            },
            async submit_click() {
                await this.application.user.update(this.request.plain)
                this.application.infos.push(this.$t('profile.successMessage'))
                await this.application.setSection(container.application.constructor.Section.Main)
            },
            /**
             *
             * @param {{text:string, value:string}}event
             * @returns {Promise<void>}
             */
            async locale_change(event) {
                // задаем локаль текущему пользователю
                this.application.user.locale=event
                // задаем текущую локаль приложению
                container.localeManager.currentLocale= event
                // меняем сообщения в разметке страниц
                this.$root.$options.i18n.locale = event.name
                // меняем сообщения в разметке vuetify
                this.$vuetify.lang.current = event.name
                // меняем сообщения об ошибках валидации
                localize(event.name)
                // сохраняем на сервере
                // await this.application.user.patchLocale()
            }
        },
        async created() {
            let user = await this.application.resolveUser()

            this.request = new Kopnik
            this.request.merge(this.application.user)
            if (!this.request.location || !this.request.location.lat) {
                this.request.location = {lat: 55.753215, lng: 37.622504}
            }
        },
        async mounted() {
            this.wasMessagesFromGroupAllowed = this.isMessagesFromGroupAllowed = await this.application.user.isMessagesFromGroupAllowed()
            // https://vk.com/dev/widget_allow_messages_from_community
            if (!this.isMessagesFromGroupAllowed) {
                container.VK.Widgets.AllowMessagesFromCommunity("vk_allow_messages_from_community", {height: 30}, 144968351);
                container.VK.Observer.subscribe("widgets.allowMessagesFromCommunity.allowed", (userId) => {
                    this.isMessagesFromGroupAllowed = true
                })
                container.VK.Observer.subscribe("widgets.allowMessagesFromCommunity.denied", (userId) => {
                    this.isMessagesFromGroupAllowed = false
                })
            }
        },
        beforeDestroy() {
            container.VK.Observer.unsubscribe("widgets.allowMessagesFromCommunity.allowed")
            container.VK.Observer.unsubscribe("widgets.allowMessagesFromCommunity.denied")
        },
    }
</script>
