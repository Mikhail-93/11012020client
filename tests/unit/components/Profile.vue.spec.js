import flushPromises from "flush-promises";
import Vue from 'vue'
import { mount } from '@vue/test-utils'

import vuePlugins from "../../test-setup";
import KTen from "../../../src/components/KTen";
import {bottle, container} from "../../../src/bottle/bottle";
import {Kopnik} from "../../../src/models";
import ProfileVue from "../../../src/components/ProfileVue";
import Application from "../../../src/application/Application";

describe('unit components Profile', () => {
    beforeAll(async () => {
    })

    beforeEach(() => {
    })

    it('draw new', async () => {
        await login(2)
        const kopnik= Kopnik.getReference(2)
        await container.application.authenticate()
        const wrapper = mount(ProfileVue, {
            ...vuePlugins,
        })
        await flushPromises()
        // expect(wrapper.vm.$el).toMatchSnapshot()
    })

    it('submit', async () => {
        await login(2)
        const kopnik= Kopnik.getReference(2)
        await container.application.authenticate()
        const wrapper = mount(ProfileVue, {
            ...vuePlugins,
        })
        const submitBtn= wrapper.find('#submit')
        submitBtn.trigger('click')
        await flushPromises()
        expect(container.application.section).toBe(Application.Section.Main)
    })
})
