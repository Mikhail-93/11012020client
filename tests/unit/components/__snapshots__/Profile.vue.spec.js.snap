// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit components Profile draw new 1`] = `
<div
  class="container fill-height container--fluid"
>
  <div
    class="mx-auto"
    style="width: 100%; max-width: 350px;"
  >
    <div
      class="v-card v-sheet theme--light elevation-12"
    >
      <div
        class="v-card__text"
      >
        <form
          class="v-form"
          novalidate="novalidate"
        >
          <div
            class="v-list v-sheet v-sheet--tile theme--light"
            role="list"
          >
            <div
              class="v-list-item theme--light"
              role="listitem"
              tabindex="-1"
            >
              <span
                class="v-badge {avatarMxAuto: 'mx-auto'} v-badge--bottom theme--light"
              >
                <div
                  class="v-avatar"
                  style="height: 128px; min-width: 128px; width: 128px;"
                >
                  <img
                    src="https://lorempixel.com/200/200/?38194"
                    style="object-fit: cover;"
                  />
                </div>
                <span
                  class="v-badge__wrapper"
                >
                  <span
                    aria-atomic="true"
                    aria-label="знак"
                    aria-live="polite"
                    class="v-badge__badge orange"
                    role="status"
                  >
                    1
                  </span>
                </span>
              </span>
            </div>
             
            <!---->
             
            <div
              class="v-list-item theme--light"
              role="listitem"
              tabindex="-1"
            >
              <div
                class="v-list-item__content"
              >
                <div
                  class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select v-autocomplete"
                >
                  <div
                    class="v-input__control"
                  >
                    <div
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-owns="list-16"
                      class="v-input__slot"
                      role="combobox"
                    >
                      <div
                        class="v-select__slot"
                      >
                        <label
                          class="v-label v-label--active theme--light"
                          for="input-16"
                          style="left: 0px; position: absolute;"
                        >
                          Язык / Language
                        </label>
                        <input
                          autocomplete="off"
                          id="input-16"
                          type="text"
                        />
                        <div
                          class="v-input__append-inner"
                        >
                          <div
                            class="v-input__icon v-input__icon--append"
                          >
                            <i
                              aria-hidden="true"
                              class="v-icon notranslate mdi mdi-menu-down theme--light"
                            />
                          </div>
                        </div>
                        <input
                          type="hidden"
                          value="ru"
                        />
                      </div>
                      <div
                        class="v-menu"
                      >
                        <div
                          class="v-menu__content theme--light v-autocomplete__content"
                          style="max-height: 304px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                        />
                      </div>
                    </div>
                    <div
                      class="v-text-field__details"
                    >
                      <div
                        class="v-messages theme--light"
                      >
                        <div
                          class="v-messages__wrapper"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-list-item theme--light"
              role="listitem"
              tabindex="-1"
            >
              <div
                class="v-list-item__content"
              >
                <span>
                  <div
                    class="v-input v-input--has-state v-input--is-label-active v-input--is-dirty theme--light v-text-field success--text"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot"
                      >
                        <div
                          class="v-text-field__slot"
                        >
                          <label
                            class="v-label v-label--active theme--light success--text"
                            for="input-24"
                            style="left: 0px; position: absolute;"
                          >
                            Фамилия
                          </label>
                          <input
                            id="input-24"
                            type="text"
                          />
                        </div>
                      </div>
                      <div
                        class="v-text-field__details"
                      >
                        <div
                          class="v-messages theme--light success--text"
                        >
                          <div
                            class="v-messages__wrapper"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </span>
              </div>
            </div>
             
            <div
              class="v-list-item theme--light"
              role="listitem"
              tabindex="-1"
            >
              <div
                class="v-list-item__content"
              >
                <span>
                  <div
                    class="v-input v-input--has-state v-input--is-label-active v-input--is-dirty theme--light v-text-field success--text"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot"
                      >
                        <div
                          class="v-text-field__slot"
                        >
                          <label
                            class="v-label v-label--active theme--light success--text"
                            for="input-29"
                            style="left: 0px; position: absolute;"
                          >
                            Имя
                          </label>
                          <input
                            id="input-29"
                            type="text"
                          />
                        </div>
                      </div>
                      <div
                        class="v-text-field__details"
                      >
                        <div
                          class="v-messages theme--light success--text"
                        >
                          <div
                            class="v-messages__wrapper"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </span>
              </div>
            </div>
             
            <div
              class="v-list-item theme--light"
              role="listitem"
              tabindex="-1"
            >
              <div
                class="v-list-item__content"
              >
                <span>
                  <div
                    class="v-input v-input--has-state v-input--is-label-active v-input--is-dirty theme--light v-text-field success--text"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot"
                      >
                        <div
                          class="v-text-field__slot"
                        >
                          <label
                            class="v-label v-label--active theme--light success--text"
                            for="input-34"
                            style="left: 0px; position: absolute;"
                          >
                            Отчество
                          </label>
                          <input
                            id="input-34"
                            type="text"
                          />
                        </div>
                      </div>
                      <div
                        class="v-text-field__details"
                      >
                        <div
                          class="v-messages theme--light success--text"
                        >
                          <div
                            class="v-messages__wrapper"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </span>
              </div>
            </div>
             
            <div
              class="v-list-item theme--light"
              role="listitem"
              tabindex="-1"
            >
              <div
                class="v-list-item__content"
              >
                <span>
                  <div
                    class="v-input v-input--has-state v-input--is-label-active v-input--is-dirty theme--light v-text-field success--text"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot"
                      >
                        <div
                          class="v-text-field__slot"
                        >
                          <label
                            class="v-label v-label--active theme--light success--text"
                            for="input-39"
                            style="left: 0px; position: absolute;"
                          >
                            Год рождения
                          </label>
                          <input
                            id="input-39"
                            type="text"
                          />
                        </div>
                      </div>
                      <div
                        class="v-text-field__details"
                      >
                        <div
                          class="v-messages theme--light success--text"
                        >
                          <div
                            class="v-messages__wrapper"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </span>
              </div>
            </div>
             
            <div
              class="v-list-item theme--light"
              role="listitem"
              tabindex="-1"
            >
              <div
                class="v-list-item__content"
              >
                <span>
                  <div
                    class="v-input theme--light v-text-field"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot"
                      >
                        <div
                          class="v-text-field__slot"
                        >
                          <label
                            class="v-label theme--light"
                            for="input-44"
                            style="left: 0px; position: absolute;"
                          >
                            Последние 4 цифры пасспорта
                          </label>
                          <input
                            id="input-44"
                            type="text"
                          />
                        </div>
                      </div>
                      <div
                        class="v-text-field__details"
                      >
                        <div
                          class="v-messages theme--light"
                        >
                          <div
                            class="v-messages__wrapper"
                          />
                        </div>
                        <div
                          class="v-counter theme--light"
                        >
                          0 / 4
                        </div>
                      </div>
                    </div>
                  </div>
                </span>
              </div>
            </div>
             
            <div
              class="v-list-item theme--light"
              role="listitem"
              tabindex="-1"
            >
              <div
                class="v-list-item__content"
              >
                <div
                  class="vue2leaflet-map leaflet-container leaflet-grab leaflet-touch-drag"
                  style="z-index: 0; height: 50vh; position: relative;"
                  tabindex="0"
                >
                  <div />
                  <div />
                  <div />
                   
                  <!---->
                  <div
                    class="leaflet-pane leaflet-map-pane"
                    style="left: 0px; top: 0px;"
                  >
                    <div
                      class="leaflet-pane leaflet-tile-pane"
                    >
                      <div
                        class="leaflet-layer "
                        style="z-index: 1;"
                      >
                        <div
                          class="leaflet-tile-container leaflet-zoom-animated"
                          style="z-index: 18; left: 0px; top: 0px;"
                        >
                          <img
                            alt=""
                            class="leaflet-tile"
                            role="presentation"
                            src="https://b.tile.openstreetmap.org/14/9904/5121.png"
                            style="width: 256px; height: 256px; left: -62px; top: -228px;"
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      class="leaflet-pane leaflet-shadow-pane"
                    >
                      <img
                        alt=""
                        class="leaflet-marker-shadow leaflet-zoom-hide"
                        src="marker-shadow.png"
                        style="margin-left: -12px; margin-top: -41px; width: 41px; height: 41px; left: 0px; top: 0px;"
                      />
                    </div>
                    <div
                      class="leaflet-pane leaflet-overlay-pane"
                    />
                    <div
                      class="leaflet-pane leaflet-marker-pane"
                    >
                      <img
                        alt=""
                        class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                        src="marker-icon.png"
                        style="margin-left: -12px; margin-top: -41px; width: 25px; height: 41px; left: 0px; top: 0px; z-index: 0;"
                        tabindex="0"
                      />
                    </div>
                    <div
                      class="leaflet-pane leaflet-tooltip-pane"
                    />
                    <div
                      class="leaflet-pane leaflet-popup-pane"
                    />
                  </div>
                  <div
                    class="leaflet-control-container"
                  >
                    <div
                      class="leaflet-top leaflet-left"
                    >
                      <div
                        class="leaflet-control-zoom leaflet-bar leaflet-control"
                      >
                        <a
                          aria-label="Zoom in"
                          class="leaflet-control-zoom-in"
                          href="#"
                          role="button"
                          title="Zoom in"
                        >
                          +
                        </a>
                        <a
                          aria-label="Zoom out"
                          class="leaflet-control-zoom-out"
                          href="#"
                          role="button"
                          title="Zoom out"
                        >
                          −
                        </a>
                      </div>
                    </div>
                    <div
                      class="leaflet-top leaflet-right"
                    />
                    <div
                      class="leaflet-bottom leaflet-left"
                    />
                    <div
                      class="leaflet-bottom leaflet-right"
                    >
                      <div
                        class="leaflet-control-locate leaflet-bar leaflet-control"
                      >
                        <a
                          class="leaflet-bar-part leaflet-bar-part-single"
                          title="Определить мое местоположение"
                        >
                          <span
                            class="fa fa-map-marker"
                          />
                        </a>
                      </div>
                      <div
                        class="leaflet-control-attribution leaflet-control"
                      >
                        <a
                          href="https://leafletjs.com"
                          title="A JS library for interactive maps"
                        >
                          Leaflet
                        </a>
                         | Map data: © 
                        <a
                          href="http://www.openstreetmap.org/copyright"
                        >
                          OpenStreetMap
                        </a>
                        , 
                        <a
                          href="http://viewfinderpanoramas.org"
                        >
                          SRTM
                        </a>
                         | Map style: © 
                        <a
                          href="https://opentopomap.org"
                        >
                          OpenTopoMap
                        </a>
                         (
                        <a
                          href="https://creativecommons.org/licenses/by-sa/3.0/"
                        >
                          CC-BY-SA
                        </a>
                        )
                      </div>
                    </div>
                  </div>
                   
                  <!---->
                   
                  <div
                    style="display: none;"
                  />
                   
                  <!---->
                </div>
              </div>
            </div>
             
          </div>
           
          <button
            class="v-btn v-btn--block v-btn--contained theme--light v-size--default primary"
            type="button"
          >
            <span
              class="v-btn__content"
            >
              
                        На подтверждение
                    
            </span>
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
